server/src/main/java/org/elasticsearch/common/time/FormatNames.java

```diff
@@ -0,0 +1,124 @@
+    STRICTBASICWEEKDATETIME("strictBasicWeekDateTime", "strict_basic_week_date_time"),
+    STRICTBASICWEEKDATETIMENOMILLIS("strictBasicWeekDateTimeNoMillis", "strict_basic_week_date_time_no_millis"),
+    STRICTDATE("strictDate", "strict_date"),
+    STRICTDATEHOUR("strictDateHour", "strict_date_hour"),
+    STRICTDATEHOURMINUTE("strictDateHourMinute", "strict_date_hour_minute"),
+    STRICTDATEHOURMINUTESECOND("strictDateHourMinuteSecond", "strict_date_hour_minute_second"),
+    STRICTDATEHOURMINUTESECONDFRACTION("strictDateHourMinuteSecondFraction", "strict_date_hour_minute_second_fraction"),
+    STRICTDATEHOURMINUTESECONDMILLIS("strictDateHourMinuteSecondMillis", "strict_date_hour_minute_second_millis"),
+    STRICTDATEOPTIONALTIME("strictDateOptionalTime", "strict_date_optional_time"),
+    STRICTDATEOPTIONALTIMENANOS("strictDateOptionalTimeNanos", "strict_date_optional_time_nanos"),
+    STRICTDATETIME("strictDateTime", "strict_date_time"),
+    STRICTDATETIMENOMILLIS("strictDateTimeNoMillis", "strict_date_time_no_millis"),
+    STRICTHOUR("strictHour", "strict_hour"),
+    STRICTHOURMINUTE("strictHourMinute", "strict_hour_minute"),
+    STRICTHOURMINUTESECOND("strictHourMinuteSecond", "strict_hour_minute_second"),
+    STRICTHOURMINUTESECONDFRACTION("strictHourMinuteSecondFraction", "strict_hour_minute_second_fraction"),
+    STRICTHOURMINUTESECONDMILLIS("strictHourMinuteSecondMillis", "strict_hour_minute_second_millis"),
+    STRICTORDINALDATE("strictOrdinalDate", "strict_ordinal_date"),
+    STRICTORDINALDATETIME("strictOrdinalDateTime", "strict_ordinal_date_time"),
+    STRICTORDINALDATETIMENOMILLIS("strictOrdinalDateTimeNoMillis", "strict_ordinal_date_time_no_millis"),
+    STRICTTIME("strictTime", "strict_time"),
+    STRICTTIMENOMILLIS("strictTimeNoMillis", "strict_time_no_millis"),
+    STRICTTTIME("strictTTime", "strict_t_time"),
+    STRICTTTIMENOMILLIS("strictTTimeNoMillis", "strict_t_time_no_millis"),
+    STRICTWEEKDATE("strictWeekDate", "strict_week_date"),
+    STRICTWEEKDATETIME("strictWeekDateTime", "strict_week_date_time"),
+    STRICTWEEKDATETIMENOMILLIS("strictWeekDateTimeNoMillis", "strict_week_date_time_no_millis"),
+    STRICTWEEKYEAR("strictWeekyear", "strict_weekyear"),
+    STRICTWEEKYEARWEEK("strictWeekyearWeek", "strict_weekyear_week"),
+    STRICTWEEKYEARWEEKDAY("strictWeekyearWeekDay", "strict_weekyear_week_day"),
+    STRICTYEAR("strictYear", "strict_year"),
+    STRICTYEARMONTH("strictYearMonth", "strict_year_month"),
+    STRICTYEARMONTHDAY("strictYearMonthDay", "strict_year_month_day");
+
+    final String camelCaseName;
+    final String snakeCaseName;
+
+    FormatNames(String camelCaseName, String snakeCaseName) {
+        this.camelCaseName = camelCaseName;
+        this.snakeCaseName = snakeCaseName;
+    }
+
+    public static boolean exist(String format) {
+        for(FormatNames name : values()){
+            if(name.camelCaseName.equals(format) || name.snakeCaseName.equals(format))
```

Comment:
How often is this method going to be called?  If often then it's probably worth putting the strings in a hash set as there are lots of them and a non-match will have to compare every single one as this stands.

Reply:
good idea, there is more then 120 of them so definitely worth putting into a set